File: backend/src/main/java/com/example/swp/features/submission/SubmissionService.java
package com.example.swp.features.submission;

import com.example.swp.features.submission.dto.request.CreateSubmissionRequest;
import com.example.swp.features.submission.dto.response.SubmissionResponse;

import java.util.List;

public interface SubmissionService {
    SubmissionResponse createSubmission(CreateSubmissionRequest request);
    List<SubmissionResponse> getSubmissionsByTeam(Long teamId);
    List<SubmissionResponse> getSubmissionsByRound(Long roundId);
    List<SubmissionResponse> getSubmissionsByEvent(Long eventId);
    SubmissionResponse getSubmissionById(Long id);
}

File: frontend/src/services/SubmissionService.ts
import axios from 'axios';

const API_URL = '/submissions';

export interface Submission {
    id: number;
    content: string;
    teamId: number;
    roundId: number;
    submittedAt: string;
}

export interface CreateSubmissionRequest {
    content: string;
    teamId: number;
    roundId: number;
}

const createSubmission = async (submission: CreateSubmissionRequest): Promise<Submission> => {
    const response = await axios.post(API_URL, submission, {
        headers: {
            Authorization: `Bearer ${localStorage.getItem('accessToken')}`
        }
    });
    return response.data;
};

const getSubmissionById = async (id: number): Promise<Submission> => {
    const response = await axios.get(`${API_URL}/${id}`, {
        headers: {
            Authorization: `Bearer ${localStorage.getItem('accessToken')}`
        }
    });
    return response.data;
};

const getSubmissionsByTeam = async (teamId: number): Promise<Submission[]> => {
    const response = await axios.get(`${API_URL}/team/${teamId}`, {
        headers: {
            Authorization: `Bearer ${localStorage.getItem('accessToken')}`
        }
    });
    return response.data;
};

const getSubmissionsByRound = async (roundId: number): Promise<Submission[]> => {
    const response = await axios.get(`${API_URL}/round/${roundId}`, {
        headers: {
            Authorization: `Bearer ${localStorage.getItem('accessToken')}`
        }
    });
    return response.data;
};

export const SubmissionService = {
    createSubmission,
    getSubmissionById,
    getSubmissionsByTeam,
    getSubmissionsByRound,
};